### PR TITLE
Update spec [ci: last-only]

### PR DESCRIPTION
**Pull request by adriaanm**
Be a little more discerning between "function" and "method". Update section on local returns, where this was especially confusing. Also, update spec of dynamic apply, since I was in the nabe.

### CODE REVIEW

File: spec/06-expressions.md [hunk 1]
```diff
@@ -1746,27 +1739,18 @@ a sub-expression of parameterless method type, is not evaluated in the expanded
 
 ### Dynamic Member Selection
 
-The standard Scala library defines a trait `scala.Dynamic` which defines a member
-`applyDynamic` as follows:
+The standard Scala library defines a marker trait `scala.Dynamic`. Subclasses of this trait are able to intercept selections and applications on their instances by defining methods of the names `applyDynamic`, `applyDynamicNamed`, `selectDynamic`, and `updateDynamic`. Under the conditions of [implicit conversion](#dynamic-member-selection), the following rewrites are performed, assuming $e$ has type `Dynamic`, and the originally expression does not type check under normal rules:
```
Comment (lrytz) [inline spec/06-expressions.md]: Can you maybe clarify / expand a little what "Under the conditions of implicit conversion" means?

Comment (lrytz) [inline spec/06-expressions.md]: I think "$e$ conforms to scala.Dynamic" was more spec-esque than "has type".

**Comment by adriaanm** [comment]
I polished my specese :-)

**Review by lrytz — APPROVED** [review]
Thanks!

**Comment by adriaanm** [comment]
Backported @martijnhoekstra's fix for the `return` section from 2.13.x.

**Comment by som-snytt** [comment]
You could also just change every occurrence to: "method or function or whatever."

There was a SO question where someone suggested a method is a member but a function is local.

Also, overloading resolution refers to functions, first as an application and then as the members to be resolved.

**Comment by adriaanm** [comment]
> but a function is local

that would be a local method. Maybe the terminology is historic, or the confusion historical? Or is it hysterical? As usual, the types will tell you, and methods have MethodTypes in the implementation. Functions get their own tree, but not even a dedicated type.

**Comment by som-snytt** [comment]
Next debate over whether Scala is adequate as a functional language, I'll interject that actually it's a methodical language.

**Pull request merged**